rholang/src/rust/interpreter/compiler/normalize.rs

```diff
@@ -978,249 +1060,364 @@ mod tests {
+                assert!(free_map.is_empty());
+            }),
         );
     }
 
     #[test]
     fn p_var_should_compile_as_free_var_if_its_not_in_env() {
-        let result = normalize_p_var(&p_var(), inputs());
-        assert!(result.is_ok());
-        assert_eq!(
-            result.clone().unwrap().par,
-            prepend_expr(inputs().par, new_freevar_expr(0), 0)
-        );
-
-        assert_eq!(
-            result.clone().unwrap().free_map,
-            inputs().free_map.put((
-                "x".to_string(),
-                VarSort::ProcSort,
-                SourcePosition::new(0, 0)
-            ))
+        let x = Id {
+            name: "x",
+            pos: SourcePosition::default(),
+        };
+        let proc_x = x.as_proc();
+
+        test_normalize_match_proc(
+            &proc_x,
+            defaults(),
+            test("expected to compile as free var if it's not in env", |actual_par, free_map| {
+                let expected_par = Par {
+                    exprs: vec![Expr {
+                        expr_instance: Some(ExprInstance::EVarBody(EVar {
+                            v: Some(models::rhoapi::Var {
+                                var_instance: Some(models::rhoapi::var::VarInstance::FreeVar(0)),
+                            }),
+                        })),
+                    }],
+                    ..Par::default()
+                };
+
+                assert_eq!(actual_par.exprs.len(), expected_par.exprs.len());
+                assert_eq!(free_map.len(), 1);
+            }),
         );
     }
 
     #[test]
     fn p_var_should_not_compile_if_its_in_env_of_the_wrong_sort() {
-        let bound_inputs = {
-            let mut inputs = inputs();
-            inputs.bound_map_chain = inputs.bound_map_chain.put((
-                "x".to_string(),
-                VarSort::NameSort,
-                SourcePosition::new(0, 0),
-            ));
-            inputs
+        let x = Id {
+            name: "x",
+            pos: SourcePosition::default(),
         };
+        let proc_x = x.as_proc();
 
-        let result = normalize_p_var(&p_var(), bound_inputs);
-        assert!(result.is_err());
-        assert_eq!(
-            result,
-            Err(InterpreterError::UnexpectedProcContext {
-                var_name: "x".to_string(),
-                name_var_source_position: SourcePosition::new(0, 0),
-                process_source_position: SourcePosition::new(0, 0),
-            })
-        )
+        test_normalize_match_proc(
+            &proc_x,
+            with_bindings(|bound_map| {
+                bound_map.put_id_as_name(&x);
+            }),
+            |result| {
+                assert_matches!(
+                    result,
+                    Err(InterpreterError::UnexpectedProcContext {
+                        var_name,
+                        name_var_source_position: SourcePosition::default(),
```

Comment:
A better test would use some, perhaps made up, source positions. This way we could have checked whether they are reported correctly